modules/store/src/selector.ts

```diff
@@ -122,11 +122,81 @@ export function defaultMemoize(
   return { memoized, reset, setResult, clearResult };
 }
 
+export function createSelector<State, S1, Result>(
+  s1: Selector<State, S1>,
+  projector: (s1: S1) => Result
+): MemoizedSelector<State, Result>;
+export function createSelector<State, S1, S2, Result>(
+  s1: Selector<State, S1>,
+  s2: Selector<State, S2>,
+  projector: (s1: S1, s2: S2) => Result
+): MemoizedSelector<State, Result>;
+export function createSelector<State, S1, S2, S3, Result>(
+  s1: Selector<State, S1>,
+  s2: Selector<State, S2>,
+  s3: Selector<State, S3>,
+  projector: (s1: S1, s2: S2, s3: S3) => Result
+): MemoizedSelector<State, Result>;
+export function createSelector<State, S1, S2, S3, S4, Result>(
+  s1: Selector<State, S1>,
+  s2: Selector<State, S2>,
+  s3: Selector<State, S3>,
+  s4: Selector<State, S4>,
+  projector: (s1: S1, s2: S2, s3: S3, s4: S4) => Result
+): MemoizedSelector<State, Result>;
+export function createSelector<State, S1, S2, S3, S4, S5, Result>(
+  s1: Selector<State, S1>,
+  s2: Selector<State, S2>,
+  s3: Selector<State, S3>,
+  s4: Selector<State, S4>,
+  s5: Selector<State, S5>,
+  projector: (s1: S1, s2: S2, s3: S3, s4: S4, s5: S5) => Result
+): MemoizedSelector<State, Result>;
+export function createSelector<State, S1, S2, S3, S4, S5, S6, Result>(
+  s1: Selector<State, S1>,
+  s2: Selector<State, S2>,
+  s3: Selector<State, S3>,
+  s4: Selector<State, S4>,
+  s5: Selector<State, S5>,
+  s6: Selector<State, S6>,
+  projector: (s1: S1, s2: S2, s3: S3, s4: S4, s5: S5, s6: S6) => Result
+): MemoizedSelector<State, Result>;
+export function createSelector<State, S1, S2, S3, S4, S5, S6, S7, Result>(
+  s1: Selector<State, S1>,
+  s2: Selector<State, S2>,
+  s3: Selector<State, S3>,
+  s4: Selector<State, S4>,
+  s5: Selector<State, S5>,
+  s6: Selector<State, S6>,
+  s7: Selector<State, S7>,
+  projector: (s1: S1, s2: S2, s3: S3, s4: S4, s5: S5, s6: S6, s7: S7) => Result
+): MemoizedSelector<State, Result>;
+export function createSelector<State, S1, S2, S3, S4, S5, S6, S7, S8, Result>(
+  s1: Selector<State, S1>,
+  s2: Selector<State, S2>,
+  s3: Selector<State, S3>,
+  s4: Selector<State, S4>,
+  s5: Selector<State, S5>,
+  s6: Selector<State, S6>,
+  s7: Selector<State, S7>,
+  s8: Selector<State, S8>,
+  projector: (
+    s1: S1,
+    s2: S2,
+    s3: S3,
+    s4: S4,
+    s5: S5,
+    s6: S6,
+    s7: S7,
+    s8: S8
+  ) => Result
+): MemoizedSelector<State, Result>;
+
 export function createSelector<State, Slices extends unknown[], Result>(
-  ...args: [...Selector<State, unknown>[], unknown] &
+  ...args: [...slices: Selector<State, unknown>[], projector: unknown] &
     [
-      ...{ [i in keyof Slices]: Selector<State, Slices[i]> },
-      (...s: Slices) => Result
+      ...slices: { [i in keyof Slices]: Selector<State, Slices[i]> },
+      projector: (...s: Slices) => Result
```

Comment:
I added names to clarify what these tuples represented.

Reply:
Nice, I didn't know we could do that! 🤩

Reply:
It's quite handy bc variadic tuple types become quite cumbersome looking quickly 😅